readthedocs/search/parsers.py

```diff
@@ -140,21 +140,68 @@ def _parse_sections(self, title, body):
         except Exception as e:
             log.info('Unable to index section', section=str(e))
 
+        # All terms in dls are treated as sections.
+        # After indexing them, we remove them from the body such that their contents aren't
+        # repeated in the following indexing
+        dls = body.css("dl")
+        dl_nodes_to_be_removed = []
+        for dl in dls:
+            # Select all dts with id defined
+            dts = dl.css('dt[id]:not([id=""])')
+
+            # https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dt
+            # multiple <dt> elements in a row indicate several terms that are
+            # all defined by the immediate next <dd> element.
+            for dt in dts:
+                if dl not in dl_nodes_to_be_removed:
+                    dl_nodes_to_be_removed.append(dl)
+                try:
+                    title, _id = self._parse_dt(dt)
+                    # https://developer.mozilla.org/en-US/docs/Web/CSS/General_sibling_combinator
+                    dd = dt.css_first("dt ~ dd")
+                    # We only index a dt with an id attribute and an accompanying dd
+                    if not dd or not _id:
+                        continue
+
+                    # The content of the <dt> section is the content of the accompanying <dd>
+                    content = dd.text()
```

Comment:
```suggestion
                    content = self._parse_content(dd.text())
```

Reply:
This didn't work, it gives me a bunch of attribute errors.. I think it doesn't work on strings.

I'm not sure, but I think that the previous method is fine, it returns all nested text nodes.